lib/ace/mode/latex_parser.js

```diff
@@ -1,30 +1,59 @@
 define(function(require, exports, module) {
 "use strict";
 
-var WORD_PATTERN = "[a-zA-Zа-яА-ЯёЁ]+";
-var WORD_REGEX = new RegExp("^" + WORD_PATTERN);
+var WORD_REGEX = /^[a-zа-яё]+/i,
+    // % * <foo@bar.com> 2015-01-12T00:17:27.829Z:
+    DISCUSSION_COMMENT_REGEXP = /^\s*%\s([\*\^])\s<(.+)>(\s(.+))?:$/,
+    DISCUSSION_TEXT_ROW_REGEXP = /^\s*%\s*(.*)/;
 
 var LatexParser = exports.LatexParser = function(bgTokenizer) {
     this.myErrors = [];
+    this.myDiscussions = [];
     this.myTokenizer = bgTokenizer;
 };
 
 (function(){
+    this.setSpellingCheckDictionary = function(dictionary) {
+        this.myDictionary = dictionary;
+    };
 
-    this.go = function(doc, dictionary){
-        if (dictionary == null) {
+    /**
+     * Parse the document
+     * @param doc              document object
+     * @param doSpellcheck     true if we want spellcheck
+     * @param doGetDiscussions true if we want get text comments
+     */
+    this.go = function(doc, doSpellcheck, doGetDiscussions) {
+        if (doSpellcheck && this.myDictionary === null) {
+            // can't spellcheck without dictionary
+            doSpellcheck = false;
+        }
+        if (!doSpellcheck && !doGetDiscussions) {
+            // there is nothing to do
             return;
         }
+
         var lines = doc.getAllLines();
 
         this.myErrors = [];
+        this.myDiscussions = [];
+        this.myLastReadDiscussionComment = null;
 
         for (var row = 0, linesLength = lines.length; row < linesLength; ++row) {
             var line = lines[row];
             var tokens = this.myTokenizer.getTokens(row);
 
             var column = 0;
-            for (var j = 0, tokenLength = tokens.length; j < tokenLength; ++j) {
+            var tokenLength = tokens.length;
+            // stop parse discussion on empty line
+            // otherwise we will add a regular comment to the discussion:
+            //     % DISCUSSION
+            //     EMPTY_LINE
+            //     % REGULAR COMMENT
+            if (tokenLength === 0) {
+              doGetDiscussions && saveLastReadDiscussionComment.call(this);
```

Comment:
what's the point of calling the function using `call(this)`? If you call the fxn like `fxn()` then the context will the this, no?